### ISSUE TITLE
[BUG] l.e.h.HandleFieldNameConstants injects field members only after it populates class initializer, causing enum values array to be empty.

### ISSUE THREAD

**Issue by kschubert** [rzwitserloot/lombok#2552]
`lombok.eclipse.handlers.HandleFieldNameConstants`, in `createInnerTypeFieldNameConstants(..)`, [injects the class initializer method (<clinit>)](https://github.com/rzwitserloot/lombok/blob/7dfbe4323c15cbd88983380b27a250d2a381d148/src/core/lombok/eclipse/handlers/HandleFieldNameConstants.java#L176), [generates the method's body](https://github.com/rzwitserloot/lombok/blob/7dfbe4323c15cbd88983380b27a250d2a381d148/src/core/lombok/eclipse/handlers/HandleFieldNameConstants.java#L177), and only then [injects the field members](https://github.com/rzwitserloot/lombok/blob/7dfbe4323c15cbd88983380b27a250d2a381d148/src/core/lombok/eclipse/handlers/HandleFieldNameConstants.java#L180). The generated class intializer method body contains the code to initialize and populate the synthetic enum values array - but since no field members have been injected at the time it is generated, the array is initialized to be of size 0/empty.

**To Reproduce**
```
@FieldNameConstants(asEnum = true)
public class Example {

    String aField;


    public static void main(String[] args) {
        System.out.println(Example.Fields.values().length);
    }
}
```

Compiled under Eclipse, this application prints out "0" instead of "1".

**Expected behavior**
The class initializer method's body should only be generated after the field members have been injected.

**Version info:**
 - Lombok version: 1.18.12
 - Platform: Eclipse 2020-06 (4.16.0) - Lombok v1.18.12

**Comment by kschubert** [comment]
I solved the issue for me through the changes from [PR 2553](https://github.com/rzwitserloot/lombok/pull/2553).

**Comment by Rawi01** [comment]
Hmm, this seems to be a duplicate of https://github.com/rzwitserloot/lombok/issues/1969 and I thought I fixed that one a few months ago. I also tried out your example using the edge version and that seems to work.

**Comment by kschubert** [comment]
I'm sorry - what is "the edge version" ?

The Eclipse I'm using says that the version of the "Eclipse Java Development Tools/org.eclipse.jdt.feature.group" is 3.18.400.v20200604-0540

I'm definitely seeing a "0" instead of a "1". And [PR 2553](https://github.com/rzwitserloot/lombok/pull/2553) definitely fixes it.

**Comment by randakar** [comment]
https://projectlombok.org/download-edge

**Comment by kschubert** [comment]
Lombok Edge fixes the issue :smiley:
